task6/task.js

```diff
@@ -0,0 +1,287 @@
+   	hashTags: ['#gelo','#ni'],
+   	likes: ['max', 'Иванов Иван']
+	},
+	{
+		id: '17',
+   	description: 'фЫврш гффы фы тв ргц фыв',
+   	createdAt: new Date('2020-03-22T23:00:00'),
+   	author: 'Да да я',
+   	photoLink: 'images/1.png',
+   	hashTags: ['#gelo','#ni'],
+   	likes: ['max', 'Иванов Иван']
+	},
+	{
+		id: '18',
+   	description: 'фЫврш гффы фы тв ргц фыв',
+   	createdAt: new Date('2020-03-23T23:00:00'),
+   	author: 'Да да я',
+   	photoLink: 'images/1.png',
+   	hashTags: ['#gelo','#ni'],
+   	likes: ['max', 'Иванов Иван']
+	},
+	{
+		id: '19',
+   	description: 'фЫврш гффы фы тв ргц фыв',
+   	createdAt: new Date('2020-03-24T23:00:00'),
+   	author: 'Да да я',
+   	photoLink: 'images/1.png',
+   	hashTags: ['#gelo','#ni'],
+   	likes: ['max', 'Иванов Иван']
+	},
+	{
+		id: '20',
+   	description: 'фЫврш гффы фы тв ргц фыв',
+   	createdAt: new Date('2020-03-25T23:00:00'),
+   	author: 'Да да я',
+   	photoLink: 'images/1.png',
+   	hashTags: ['#gelo','#ni'],
+   	likes: ['max', 'Иванов Иван']	
+	},
+];
+let func = (function() {
+	function comp(a, b) {
+		return a.createdAt - b.createdAt;
+	}
+	return {
+		getPosts: function(skip = 0, top = 10, filterConfig ={}) {
+			let newPosts = [];
+			let filterKeys = Object.keys(filterConfig);
+			let filt;
+			if(filterKeys.length != 0)
+			{
+				filt = posts.filter(post => {
+					if(filterKeys[0] != 'hashTags')
+						return post[filterKeys[0]] == filterConfig[filterKeys[0]]
+					for(tag of filterConfig.hashTags)
+						if(post.hashTags.indexOf(tag) >= 0)
+							return true;
+					return false;
+				 });
+			}
+			else
+				filt = posts;
+			filt.sort(comp);
+			for (let i = skip; i < Math.min(skip + top, filt.length) ; i++) {
+				newPosts.push(filt[i]);
+			}
+			return newPosts;
+		},
+		getPost: function(id = '1'){
+			for(post of posts)
+				if(post.id == id)
+					return post;
+			return null;
+		},
+		validatePost: function(post){
+			if(!post.hasOwnProperty('id') || typeof(post["id"])!='string')
+				return false;
+			if(!post.hasOwnProperty('description') || typeof(post["description"])!='string' || post["description"].length > 200)
+				return false;
+			if(!post.hasOwnProperty('createdAt') || !((post["createdAt"]) instanceof Date) ||post["createdAt"]=='Invalid Date')
+				return false;
+			if(!post.hasOwnProperty('author') || typeof(post["author"])!='string' || post["author"].length==0)
+				return false;
+			if(!post.hasOwnProperty('likes') || !Array.isArray(post["likes"]))
+				return false;
+			return true;
+		},
+		addPost: function(post){
+			if(func.validatePost(post)){
+				posts.push(post);
+				return true;
+			}
+			return false;
+		},
+		editPost: function(id,post){
+			let ind = 0;
+			for(let i = 0; i < posts.length;i++)
+				if(posts[i].id == id)
+					{
+						ind = i;
+						break;
+					}
+			let editPost = posts[ind];
+			for(let key in post){
+				if(key != 'id' && key != 'createdAt' && key != 'author' && key!='likes')
+					editPost[key] = post[key];
```

Comment:
Object is reference type changing it here changes it in array. So, validation can be failed, but post already changed